sbtc/src/idpack/mod.rs

```diff
@@ -0,0 +1,89 @@
+//! `idpack` placeholder module
+
+use std::hash::{DefaultHasher, Hash, Hasher};
+
+/// Placeholder for the `segmenter::Error` type
+#[derive(Debug, thiserror::Error)]
+#[error("SegmenterError")]
+pub struct SegmenterError;
+
+/// Placeholder for the `codec::SegmentEncodeError` type
+#[derive(Debug, thiserror::Error)]
+#[error("SegmentEncodeError")]
+pub struct SegmentEncodeError;
+
+/// Placeholder for the `codec::SegmentDecodeError` type
+#[derive(Debug, thiserror::Error)]
+#[error("SegmentDecodeError")]
+pub struct SegmentDecodeError;
+
+/// Placeholder for the `Segments` type
+#[derive(Debug, PartialEq, Eq)]
+pub struct Segments(Vec<u64>);
+
+impl Segments {
+    /// Create a new `Segments` instance from the given values
+    pub fn from_vec(values: Vec<u64>) -> Self {
+        Self(values)
+    }
+}
+
+/// Placeholder for the `BitmapSegmenter` type
+pub struct BitmapSegmenter;
+
+/// Placeholder for the `Segmenter` trait
+pub trait Segmenter {
+    /// Package the given values into segments
+    fn package(&self, values: &[u64]) -> Result<Segments, SegmenterError> {
+        Ok(Segments(values.to_vec()))
+    }
+    /// Estimate the size of the segments
+    fn estimate_size(&self, values: &[u64]) -> Result<usize, SegmenterError> {
```

Comment:
These two should return `Result`s either, they should always succeed.

Reply:
Can we resolve this comment? This PR will ultimately follow whatever #1486 ends up with.